### PR TITLE
Fix/caas controller machine node change

### PR DESCRIPTION
**Pull request by ycliuhw**
## Description of change

Made `get_hosts` can return empty because CaaS controller does not have machines anymore; 

## QA steps

None

## Documentation changes

None

## Bug reference

None

### CODE REVIEW

File: acceptancetests/deploy_stack.py [hunk 1]
```diff
@@ -592,11 +592,13 @@ def create_initial_model(self, upload_tools, series, boot_kwargs):
...
         """Provide the controller host."""
-        host = get_machine_dns_name(
+        hosts = {}
+        host_0 = get_machine_dns_name(
             self.client.get_controller_client(), '0')
-        if host is None:
-            raise ValueError('Could not get machine 0 host')
-        return {'0': host}
+        if host_0 is not None:
+            # Only IaaS controller has machine now.
```
Comment (wallyworld) [inline acceptancetests/deploy_stack.py]: can we add a check that this is a caas controller?

Comment (ycliuhw) [reply acceptancetests/deploy_stack.py]: as discussed, currently the controller type is not accessible from here.

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$

network issue

**Pull request merged**